### ISSUE TITLE
Installing bottles from file fails on case-sensitive file systems

### ISSUE THREAD

**Issue by dbogdanov** [Homebrew/homebrew#42470]
```
brew install essentia-HEAD.yosemite.bottle.tar.gz
Error: No such file or directory - /Users/dimi/essentia-head.yosemite.bottle.tar.gz
```
Instead of ```*HEAD*``` file, brew expects ```*head*``` file.

**Comment by mistydemeo** [comment]
This is happening within `Formualry`'s `BottleLoader`, which is invoked by the call to `ARGV.formulae`.

**Comment by mistydemeo** [comment]
Can I see the output of `brew install essentia-HEAD.yosemite.bottle.tar.gz --debug`?

**Comment by mistydemeo** [comment]
Note however that even after fixing the case-sensitivity bug, this won't work as you hope for; bottles are only supported for stable versions, and `HEAD` bottles can't be installed.

**Comment by dbogdanov** [comment]
```
brew install essentia-HEAD.yosemite.bottle.tar.gz --debug
Error: No such file or directory - /Users/dimi/essentia-head.yosemite.bottle.tar.gz
/usr/local/Library/Homebrew/formulary.rb:85:in `realpath'
/usr/local/Library/Homebrew/formulary.rb:85:in `realpath'
/usr/local/Library/Homebrew/formulary.rb:85:in `initialize'
/usr/local/Library/Homebrew/formulary.rb:213:in `new'
/usr/local/Library/Homebrew/formulary.rb:213:in `loader_for'
/usr/local/Library/Homebrew/formulary.rb:174:in `factory'
/usr/local/Library/Homebrew/extend/ARGV.rb:16:in `block in formulae'
/usr/local/Library/Homebrew/extend/ARGV.rb:16:in `map'
/usr/local/Library/Homebrew/extend/ARGV.rb:16:in `formulae'
/usr/local/Library/Homebrew/cmd/install.rb:41:in `install'
/usr/local/Library/brew.rb:127:in `<main>'
```

**Comment by dbogdanov** [comment]
Hmm, I think I was able to install a ```HEAD```bottle using ```--HEAD``` flag.

**Comment by mistydemeo** [comment]
The name is intentionally downcased here: https://github.com/Homebrew/homebrew/blob/5a2292ff22d08b6c8b7c06bb15d6812577c977d2/Library/Homebrew/extend/ARGV.rb#L16

```ruby
  def formulae
    require "formula"
    @formulae ||= (downcased_unique_named - casks).map { |name| Formulary.factory(name, spec) }
  end
```

The intention is that, formula filenames always being lowercase, anything that doesn't appear to be a path on disk or a URL should be converted to a canonical representation. This ensures that, for example, `brew info Gist` works as well as `brew info gist`.

I assume `brew info /Users/dimi/essentia-head.yosemite.bottle.tar.gz` works?

**Comment by mistydemeo** [comment]
Should be fixed by 66e2550bc233ed9914e6e6f9166f5b054cd891e9, can you test again?
